### PR TITLE
feat: v2 and v3 in same release

### PR DESCRIPTION
**Pull request by subzero10**
## Status
**READY**

## Description
Support vue 2.x and 3.x in the same release.

Note: ~~This needs to be a major version bump because we are changing the way to initialize the component. Though I am still looking into it.~~ UPDATE: I was able to make it backwards compatible by changing the `init` method to `install`. `install` is called both from vue2.x with `Vue.use()` and from vue3.x with `app.use()`.

Features:
- One release to support both vue2.x and vue3.x.
- `vue` and `@honeybadger-io/js` are now peer dependencies.
- 2 new example projects (vue2.x and vue3.x).
- Removed e2e tests. We should be covered with unit tests.
- Removed demo project because it was outdated and the structure was confusing (it was mixed with the actual integration code). All webpack references were for demo project and e2e tests.
- Unit tests only run against vue3.x. It was too messy to add unit tests for vue2.x (needed to do stuff like [this](https://dev.to/joshx/test-your-npm-package-against-multiple-versions-of-its-peer-dependency-34j4) for many packages) and the code was too dirty.
- Fix unpkg.com - now it reports: `Cannot find "/dist/honeybadger-vue.min.js" in @honeybadger-io/vue@1.1.0`

## Todos
- [x] Honeybadger `init` method that takes a Vue instance (vue 2.x) or an app instance (vue 3.x)
- [x] Type Definitions
- [x] Make `vue` devDep as a peerDep.
- [x] Make sure tests work for vue 3.x (optionally for vue 2.x)
- [x] Vue2.x example
- [x] Vue3.x example
- [x] Manually test with _vue2.x on master_ vs _vue2.x on this branch_
- [x] Manually test with _vue3.x on vue3 branch_ vs _vue3.x on this branch_
- [x] Remove e2e tests
- [x] Documentation - Update vue 3.x page to point to master branch. Update readme with new instructions to install plugin.
- [x] Support v2 and v3 without breaking changes
- [x] Fix rollup to build minified file for unpkg.com

### CODE REVIEW

**Comment by subzero10** [comment]
OK, I think this is ready. It's a big PR, and there are still things I left out (generating type definitions automatically, unit tests for vue2.x). Nevertheless, I am pleased with the result.

File: build/rollup.config.js [hunk 1]
```diff
@@ -1,31 +0,0 @@
-import buble from 'rollup-plugin-buble'; // Transpile/polyfill with reasonable browser support
```
Comment (subzero10) [inline build/rollup.config.js:1]: Moved to root folder, since all other files from this folder were deleted.

**Comment by subzero10** [comment]
Notable limitations:
- Typescript support may not work with some versions of vue 2.x.  I don't have a way to make the `.d.ts` support both releases yet, but I'm still looking.
- Error logging is not as rich in vue 3.x as it is in vue 2.x. This is discussed in #881.

**Review by joshuap — APPROVED** [review]
LGTM, nice work!

I take it no one is starting *new* vue2 projects these days, right? The people who may need the vue2 support moving forward will be folks who haven't upgraded yet? If there are issues w/ typescript support + vue2, we could recommend that those vue2+TS users lock their package.json to the older @honeybadger-io/vue version before this release.

**Comment by joshuap** [comment]
Adding Shalvah as a reviewer just in case he has time to weigh in.

**Comment by subzero10** [comment]
> I take it no one is starting _new_ vue2 projects these days, right? The people who may need the vue2 support moving forward will be folks who haven't upgraded yet? 

Correct. vue3 is now the stable release.

> If there are issues w/ typescript support + vue2, we could recommend that those vue2+TS users lock their package.json to the older @honeybadger-io/vue version before this release.

👍

**Comment by shalvah** [comment]
Ah, sorry. Not much I can contribute, as I don't do enough frontend to know the changes in Vue 3.

**Comment by subzero10** [comment]
> Ah, sorry. Not much I can contribute, as I don't do enough frontend to know the changes in Vue 3.

Actully, if you take a look at the [code](https://github.com/honeybadger-io/honeybadger-vue/pull/974/files#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R41-R49), Vue 2 + Vue 3 compatibility is less than 5 lines of code. However, the current project structure made it difficult to support both of them and I had to do a few changes for that. While I was on it, I discovered a few other points that could be improved so I did them all in one go (it reduced the complexity of my original goals).

**Pull request merged**